### DISCUSSION TITLE
Database GUI client connection mode

### DISCUSSION THREAD

**Discussion by partoche** [gravitational/teleport#21654] [Q&A] [answered]
Hello,

I have a teleport instance with auth/proxy/db_service, and a Postgres and a MySQL database i want to use.

Basically, i do this :
 - teleport login
 - teleport db login    --> Generates the client certificates i need

With the postgres database, i can connect using my GUI client using SSL and the client certificates.
With MySQL, no luck, i HAVE TO use tsh proxy db

I know that is what the documentations says (https://goteleport.com/docs/ver/11.x/connect-your-client/gui-clients/), still, it's easier to configure the GUI client to do the SSL part, as i don't have to change the local port each time i reconnect, i just nee to login to the proxy and the db, the certificates' path does not change.

Does anybody uses a MySQL GUI client and achieves to connect without tsh proxy db ?

I'm using MySQL workbench 8.0.32, also tried with HeidiSQL. 

Thank you.

**Comment by webvictim** [comment]
When you run `tsh db login <mysql-db>`, what does `tsh db config` say?

Which parts of this are different to `tsh db config` after running `tsh db login <postgres-db>`?

Also FYI, you can `tsh proxy db <mysql-db> --port 12345` to keep the port static if desired and avoid changing it.

**Comment by partoche** [comment, marked as answer]
Hello,

Thank you for your reply.

The only thing that differs between Postgres and MySQL is the port when i run tsh db config --format=yaml.
My teleport instance only allow incoming port 443, and is not listening on 3036.

Postgres config shows 443
ca: C:\Users\bob\.tsh\keys\teleport.example.com\cas\teleport.example.com.pem
cert: C:\Users\bob\.tsh\keys\teleport.example.com\bob@domain.com-db\teleport.example.com\postgres-srv-x509.pem
host: teleport.example.com
key: C:\Users\bob\.tsh\keys\teleport.example.com\bob@domain.com
name: postgres-srv
**port: 443**

MySQL shows port 3036 :
ca: C:\Users\bob\.tsh\keys\teleport.example.com\cas\teleport.example.com.pem
cert: C:\Users\bob\.tsh\keys\teleport.example.com\bob@domain.com-db\teleport.example.com\mariadb-srv-x509.pem
host: teleport.example.com
key: C:\Users\bob\.tsh\keys\teleport.example.com\bob@domain.com
name: mariadb-srv
**port: 3036**
user: teleportdbuser

The --port option is fine, thank you, it makes the connection process easier.

**Reply by webvictim** [reply]
That makes sense. This is jogging my memory now - I don't believe that the MySQL listener can be multiplexed on port 443 because MySQL is a server-initiated protocol rather than a client-initiated protocol (as Postgres is) - so the use of `tsh proxy db` is mandatory to connect to MySQL databases when using Teleport's single-port TLS routing/multiplexing mode.

**Reply by partoche** [reply]
Thank you for the help and information, @webvictim, it's much appreciated.
